tests/validation/tests/v3_api/test_node_annotation.py

```diff
@@ -0,0 +1,554 @@
+    node_name = node.nodeName
+
+    # get user token and client
+    token = rbac_get_user_token_by_role(role)
+    user_client = get_client_for_token(token)
+    print(cluster_detail["cluster"]["id"])
+    print(cluster_detail["cluster"])
+    cluster = user_client.list_cluster(id=cluster_detail["cluster"]["id"]).data
+    print(cluster)
+    create_kubeconfig(cluster[0])
+
+    # add annotation on node
+    command = "annotate nodes " + node_name + " " + \
+              annotation_key + "=" + annotation_value
+
+    if role == CLUSTER_OWNER:
+        execute_kubectl_cmd(command, False)
+        time.sleep(2)
+        # annotation should be added
+        validate_annotation_set_on_node(user_client, node,
+                                        annotation_key,
+                                        annotation_value)
+    elif role == CLUSTER_MEMBER:
+        result = execute_kubectl_cmd(command, False, stderr=True)
+        result = result.decode('ascii')
+        assert "cannot patch resource \"nodes\"" in result
+        assert "forbidden" in result
+    else:
+        result = execute_kubectl_cmd(command, False, stderr=True)
+        result = result.decode('ascii')
+        assert "cannot get resource \"nodes\"" in result
+        assert "forbidden" in result
+
+
+@pytest.fixture(scope='module', autouse="True")
+def create_project_client(request):
+    cluster_detail["client"], cluster_detail["cluster"] = \
+        get_user_client_and_cluster()
+
+
+def check_annotation_added(annotation):
+    def _find_condition(resource):
+        node_annotations = resource.annotations.data_dict()
+
+        if annotation in node_annotations:
+            return True
+        else:
+            return False
+
+    return _find_condition
+
+
+def validate_annotation_set_on_node(client, node,
+                                    annotation_key,
+                                    annotation_value):
+    """
+    This method checks if the annotation is
+    added on the node via API and kubectl
+    :param client: user client
+    :param node: node on which user has to validate if the annotation is added
+    :param annotation_key: annotation to be validated on the node
+    :param annotation_value: annotaton value to be checked
+    :return: None
+    """
+    print("annotaton_value: ", annotation_value)
+    # check via API
+    node = client.reload(node)
+    node_annotations = node.annotations.data_dict()
+    assert node_annotations[annotation_key] == annotation_value
+
+    # check via kubectl
+    node_name = node.nodeName
+    command = " get nodes " + node_name
+    node_detail = execute_kubectl_cmd(command)
+    print(node_detail["metadata"]["annotations"])
+    assert annotation_key in node_detail["metadata"]["annotations"], \
+        "Annotation is not set in kubectl"
+    assert node_detail["metadata"]["annotations"][annotation_key] \
+        == annotation_value
+
```

Comment:
Can this method be modified to handle multiple annotations ?

Reply:
Using this 2 times to check 2 annotations added instead.